### ISSUE TITLE
Crash with error: Thread 1: EXC_BAD_ACCESS (code=2, address=0x7fff86533e38)

### ISSUE THREAD

**Issue by vivek4292** [ephread/Instructions#226]
**Describe the bug**
 When I was using 1.2.0 and Xcode 10.2 it was working fine. As I switched to Xcode 11.2 and 1.4.0 version of Instruction. I found a crash while assigning a delegate to CoachMarksController instance.   

**To Reproduce**
Assign a delegate to an instance of CoachMarksController. In Xcode 11.2

**Expected behavior**
 It should display coach mark views smoothly. 

**Screenshots**
<img width="691" alt="Screenshot 2019-11-12 at 5 40 40 PM" src="https://user-images.githubusercontent.com/34335017/68671316-16336180-0575-11ea-8971-10e290c7b42d.png">

**Environment:**
 - Xcode: 11.2
 - Device:  e.g. iPad 9.7- inch 
 - iOS version: 13.2 
 - Dependency Manager:  Carthage

**Comment by ephread** [comment]
Hey @vivek4292, thanks for reporting the issue. I'm unable to reproduce it, would you be able to provide a sample project demonstrating the crash?

**Comment by vivek4292** [comment]
Hey, @ephread  thanks for the response. I tried this in a separate project with the same environment and it is working fine. But it's still not working in my project. 
<img width="715" alt="Screenshot 2019-11-27 at 5 58 45 PM" src="https://user-images.githubusercontent.com/34335017/69727200-0b1d2b80-1148-11ea-8587-8ea14b2b95c5.png">
In the above screenshot:
I'm instantiating an object of **CoachMarksController()** and passing to a class-level variable. 
In the next line, you can see I'm assigning a data source to it. At that moment I checked if its reference is available in memory, I found It was available(**see console**). 
When I move to the next line, I found its data source was nil. Datasource was not assigned.
Moreover when I checked its memory reference (**see console**). It was missing and the app got crashed.
Please help me.

**Comment by ephread** [comment]
> I'm instantiating an object of CoachMarksController() and passing to a class-level variable.
In the next line, you can see I'm assigning a data source to it.

This may be a trivial question, but how and where is `coacher` declared?

**Comment by vivek4292** [comment]
@ephread  Yeah it's a trivial question but I can understand in such case anyone can think of this possible reason. Agreed, it looks like an object is getting released soon. 
At first, I declared coacher at class level as lazy. 
<img width="513" alt="Screenshot 2019-12-02 at 12 17 16 PM" src="https://user-images.githubusercontent.com/34335017/69936659-6c554e00-14fe-11ea-937c-606a43baa7a4.png">
It was instantiated on first use(**not in ViewDidLoad**) and it was working fine till I was using Xcode 11.1  When I migrated this to Xcode 11.2.1 It started causing a crash. My thoughts were the same about releasing the object soon so **I tried not to use lazy, instantiating in ViedDidLoad(), even instantiate a fresh object in ViewDidLoad() and not using a class-level variable.** 
<img width="598" alt="Screenshot 2019-12-02 at 12 21 45 PM" src="https://user-images.githubusercontent.com/34335017/69936754-bcccab80-14fe-11ea-8e84-3bed89bd4fba.png">

But the crash is still there.

**Comment by kadm91** [comment]
I had the same problem with a project I am working on, I solve it by moving the code that was causing the bug to the main queue since that code was directly related to UI. this was in Xcode 13.
